### PR TITLE
Clang static analysis fix for rand()

### PR DESCRIPTION
**Pull request by utsabms**
Calling platform specific UUID function for apple to fix rand()

### CODE REVIEW

File: tests/unittests/PalTests.cpp [hunk 1]
```diff
@@ -1,370 +1,16 @@
...
 #include "common/Common.hpp"
-#include "common/MockISemanticContext.hpp"
+#include "pal/PseudoRandomGenerator.hpp"
+#include "Version.hpp"
 
 using namespace testing;
 
 class PalTests : public Test {};
 
-//---
```
Comment (sid-dahiya) [inline tests/unittests/PalTests.cpp:14]: Why are all these tests being removed?

File: tests/unittests/PalTests.cpp [hunk 2]
```diff
@@ -1,370 +1,16 @@
-#if 0
 //
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include "common/Common.hpp"
-#include "common/MockISemanticContext.hpp"
+#include "pal/PseudoRandomGenerator.hpp"
+#include "Version.hpp"
```
Comment (sid-dahiya) [inline tests/unittests/PalTests.cpp:8]: You shouldn't be using `Version.hpp` anywhere.

File: tests/unittests/PalTests.cpp (hunk 1 above)
Comment (utsabms) [reply tests/unittests/PalTests.cpp:14]: These changes were never run. The whole file was commented out. I recovered some.

Comment (sid-dahiya) [reply tests/unittests/PalTests.cpp:14]: In that case, there should be an issue tracking the tests not running. as it means there is test coverage gap right now.

File: tests/unittests/PalTests.cpp (hunk 2 above)
Comment (sid-dahiya) [reply tests/unittests/PalTests.cpp:8]: Is there a test validating SDK version? If so, this header is fine to have here. If the test is removed, you don't need this.

Comment (utsabms) [reply tests/unittests/PalTests.cpp:8]: Yep. The last one in PalTests.cpp. Thats why I added it.

File: tests/unittests/PalTests.cpp (hunk 1 above)
Comment (utsabms) [reply tests/unittests/PalTests.cpp:14]: Testcases are commented out in lib\pal\PalTests.cpp #1010

Comment (utsabms) [reply tests/unittests/PalTests.cpp:14]: I have created a bug,

**Review by lalitb — CHANGES_REQUESTED** [review]
Need more clarity on the UUID format - hex strings in upper case or lower case.

**Review by lalitb — APPROVED** [review]
Approved as discussed over teams - The format generated is similar to that for Windows so shouldn't break anything upstream. Should be good to publish once tests are passing.

**Review by raghavthind2005 — APPROVED** [review]

File: lib/pal/PAL.cpp [hunk 3]
```diff
@@ -300,6 +304,32 @@ namespace PAL_NS_BEGIN {
...
+            uuidStr = CFStringGetCStringPtr(uuidStrRef, kCFStringEncodingASCII);
+        }
+        else
+        {
+            const auto uuidNullTerminatedSize {CFStringGetLength(uuidStrRef) + 1};
+            uuidStr.resize(uuidNullTerminatedSize);
+            CFStringGetCString(uuidStrRef, &uuidStr[0], uuidNullTerminatedSize, kCFStringEncodingASCII);
+        }
+        CFRelease(uuidStrRef);
+	std::transform(uuidStr.begin(), uuidStr.end(), uuidStr.begin(), ::tolower);
```
Comment (sid-dahiya) [inline lib/pal/PAL.cpp:331]: Might be the viewer, but the indentation looks weird. Mind double checking it?

**Review by sid-dahiya — APPROVED** [review]

**Pull request merged**